docs/testing/e2e-testing-standards.md

```diff
@@ -0,0 +1,263 @@
+// raise a charge
+// this could involve calling methods or functions defined in your page objects
+await raiseCharge()
+
+// assert, verify that the action had the expected outcome 
+// confirm charge has been raised
+expect(charge).toBe('raised')
+```
+
+## Page Object Model (POM)
+
+Every page should possess one POM file to enhance maintainability and scalability of our tests. It should include all the selectors and functions which are needed to interact with the UI page or component(s).
+
+All interactions should be done using the page objects, no selectors in your tests.
+
+All assertions should be in your test, no assertion in the POM
+
+DO 👍
+
+```typescript
+// POM file './pageOobjects/foo/'
+// add all locators and functions related to the page. 
+// allowing all tests to reuse 
+
+import {expect, Locator, Page} from '@playwright/test';
+
+export class FooPage {
+    readonly errorMessage: Locator;
+
+    constructor(page: Page) {
+        this.page = page;
+        this.errorMessage = page.locator('.error-message');
+    }
+}
+
+// test file './steps/foo.ts'
+import {FooPage} from './pageObjects/foo';
+
+let fooPage: FooPage;
+
+Then(
+    'error message should be visible',
+    async function ({page}) {
+        const fooPage = new FooPage({page});
+        await expect(fooPage.errorMessage).toBeVisible();
+    }
+)
+```
+
+DO NOT ⚔️
+```typescript
+// test file './steps/foo.ts'
+// include locators directly in test
+import { Locator, Page } from '@playwright/test';
+
+Then(
+    'error message should be visible',
+    async function ({page}) {
+        await expect(page.locator('.error-message')).toBeVisible();
+    }
+)
+```
+
+## Waiting:
+
+We can avoid using any arbitrary waits until it is really, really necessary.
+
+DO NOT ⚔️
+
+```typescript
+await page.waitForTimeout(5000)
+```
+
+This can cause flaky tests as we can rarely be certain the amount of wait time is enough. It can also unnecessarily increase the test run time. Instead, we can try:
+
+DO 👍
+
+```typescript
+await page.goto(fooBarURL, {
+    waitUntil: 'domcontentloaded'
+});
+```
+
+DO 👍
+
+```typescript
+const element = page.locator("some-locator-path");
+element.waitFor({ visible: true });
+```
+
+DO 👍
+
+```typescript
+await fooPage.buttonFoo.click();
+await expect(fooPage.titlePage).toBeVisible();
+```
+
+## Selectors
+
+Avoid selectors tied to implementation and page structure.
+
+Instead, we can prioritize the below, based on [testing-library guiding principles](https://testing-library.com/docs/queries/about/#priority)
+
+- `getByRole` (this aids accessibility, reflects how users and assistive technology perceive the page)
+
+- `getByText`
+
+- `getByTestId` (add them when needed)
+
+DO NOT ⚔️
+
+```javascript
+page.locator('.opt-u > div > .summary > div:nth-child(4) > div')
+```
+
+DO 👍
+
+```javascript
+page.locator('#foo-button');
+page.getByText('OK');
+```
+
+## Naming Conventions
+
+### Files and folders
+
+Declare in **_camelCase_**
+
```

Comment:
I thought we are going to use `camelCase` for js/ts files and `kebab-case` for folders as web is using it already.

Reply:
Ok since we already the structure mentioned by you on `web`, let's go same for now. And let's get other developers view